### PR TITLE
Get rid of pyproject.toml

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by inducer** [comment]
Thanks! LGTM, feel free to merge once CI passes.

**Comment by inducer** [comment]
CI died:
```
    AttributeError: 'dict' object has no attribute '__NUMPY_SETUP__'
```

**Comment by isuruf** [comment]
Tests are failing now because of the parallel tests. Test collector doesn't know that the following two tests are the same,
```
-test/test_wrapper.py::test_unload_compiler[<pyopencl.Platform 'Portable Computing Language' at 0x7f3c548a0008>]
+test/test_wrapper.py::test_unload_compiler[<pyopencl.Platform 'Portable Computing Language' at 0x7fe8d8d30008>]
```

**Comment by inducer** [comment]
```
test/test_algorithm.py:537:1: E1102: pytest.mark.parametrize is not callable (not-callable)
```
Could you have mypy ignore `not-callable` these in `test/test_*.py`?

**Comment by isuruf** [comment]
How do I do that? Add `# pylint: disable=E1102` at the top of those files?

**Comment by inducer** [comment]
> How do I do that?

[Potential example](https://gitlab.tiker.net/inducer/pytato/-/blob/3e4bf6e3d4faaf4a348a769406b79dc7382e15ae/setup.cfg#L11)

File: pyopencl/tools.py [hunk 1]
```diff
@@ -286,8 +286,15 @@ def __str__(self):
 
             arg_values.append(tuple(arg_dict[name] for name in arg_names))
 
+    def idfn(val):
+        if isinstance(val, cl.Platform):
+            # Don't show address, so that parallel test collection works
+            return f"<pyopencl.Platform '{val.name}'>"
```
Comment (inducer) [inline pyopencl/tools.py:292]: This seems important, but I don't fully understand it. What's the connection to parallel test collection?

Comment (inducer) [reply pyopencl/tools.py:292]: OIC: https://github.com/inducer/pyopencl/runs/871508867

Comment (inducer) [reply pyopencl/tools.py:292]: Thanks!

**Comment by inducer** [comment]
I couldn't reproduce this locally with pylint 2.5.3, so I'm trying https://gitlab.tiker.net/inducer/ci-support/-/commit/e4f10003c4a980f40363328e654c0e726134a501. (However this had some other random spurious issue with `scipy.special` locally... we'll see if that crops up here.)

**Comment by inducer** [comment]
Turns out it reproduced just fine with pytest 6.0.0rc1.

> How do I do that? Add # pylint: disable=E1102 at the top of those files?

You were right. Sorry for distracting you. a32e1ca For one, I had mypy and pylint confused. :facepalm:

**Pull request merged**